onto-viewer-web-app/src/main/java/org/edmcouncil/spec/ontoviewer/webapp/controller/SearchApiController.java

```diff
@@ -25,58 +25,61 @@
+    private static final Integer DEFAULT_RESULT_PAGE = 1;
 
-  public SearchApiController(TextSearchService textSearchService,
-      OntologySearcherService ontologySearcher,
-      UpdateBlocker blocker) {
-    this.textSearchService = textSearchService;
-    this.ontologySearcher = ontologySearcher;
-    this.blocker = blocker;
-  }
-
-  @PostMapping(value = {"", "/max/{max}", "/page/{page}", "/max/{max}/page/{page}"})
-  public ResponseEntity<SearcherResult> searchJson(
-      @RequestBody String query,
-      @PathVariable Optional<Integer> max,
-      @PathVariable Optional<Integer> page) {
-    if (!blocker.isInitializeAppDone()) {
-      LOGGER.debug("Application initialization has not completed");
-      return new ResponseEntity<>(HttpStatus.SERVICE_UNAVAILABLE);
+    public SearchApiController(TextSearchService textSearchService,
+            OntologySearcherService ontologySearcher,
+            UpdateBlocker blocker) {
+        this.textSearchService = textSearchService;
+        this.ontologySearcher = ontologySearcher;
+        this.blocker = blocker;
     }
 
-    SearcherResult result;
-    try {
-      if (UrlChecker.isUrl(query)) {
-        LOGGER.info("URL detected - searching for specific element.");
+    @PostMapping(value = {"", "/max/{max}", "/page/{page}", "/max/{max}/page/{page}"})
+    public ResponseEntity<SearcherResult> searchJson(
+            @RequestBody String query,
+            @PathVariable Optional<Integer> max,
+            @PathVariable Optional<Integer> page) {
+        if (!blocker.isInitializeAppDone()) {
+            LOGGER.debug("Application initialization has not completed");
+            return new ResponseEntity<>(HttpStatus.SERVICE_UNAVAILABLE);
+        }
+        long startTimestamp = System.currentTimeMillis();
+        
+        SearcherResult result;
+        try {
+            if (UrlChecker.isUrl(query)) {
+                LOGGER.info("URL detected - searching for specific element.");
 
-        result = ontologySearcher.search(query, 0);
-      } else {
-        LOGGER.info("String detected - searching for elements with the given label.");
+                result = ontologySearcher.search(query, 0);
+            } else {
+                LOGGER.info("String detected - searching for elements with the given label.");
 
-        Integer maxResults = max.orElse(DEFAULT_MAX_SEARCH_RESULT_COUNT);
-        Integer currentPage = page.orElse(DEFAULT_RESULT_PAGE);
-        result = textSearchService.search(query, maxResults, currentPage);
-      }
+                Integer maxResults = max.orElse(DEFAULT_MAX_SEARCH_RESULT_COUNT);
+                Integer currentPage = page.orElse(DEFAULT_RESULT_PAGE);
+                result = textSearchService.search(query, maxResults, currentPage);
+                long endTimestamp = System.currentTimeMillis();
+                LOGGER.info("For query: '{}' (query time: '{}' ms) result is:\n {}", query, endTimestamp - startTimestamp, result);
```

Comment:
In `SearchController` it look `LOG.info("String detected: '{}' (query time: '{}' ms), max: '{}', page '{}', result '{}'", query, endTimestamp - startTimestamp, max, page, result);`
without above log commented by me. it is necessary to adopt one version in order to be able to search more efficiently later